blackbox/docs/appendices/release-notes/3.0.0.rst

```diff
@@ -0,0 +1,214 @@
+  ignores insert values which would cause duplicate keys.
+
+- Added support for ON CONFLICT clause in insert statements.  ``INSERT INTO ...
+  ON CONFLICT (pk_col) DO UPDATE SET col = val`` is identical to ``INSERT INTO
+  ... ON DUPLICATE KEY UPDATE col = val``.  The special EXCLUDED table can be
+  used to refer to the insert values: ``INSERT INTO ... ON CONFLICT (pk_col) DO
+  UPDATE SET col = EXCLUDED.col``
+
+- DEPRECATED: The ``ON DUPLICATE KEY UPDATE`` clause has been deprecated in
+  favor of the ``ON CONFLICT DO UPDATE SET`` clause.
+
+- Implemented the Block Hash Join algorithm which is now used for Equi-Joins.
+
+- Added new ``sys.health`` system information table to expose the health of all
+  tables and table partitions.
+
+- Added a new ``bengali`` language analyzer and a ``bengali_normalization``
+  token filter.
+
+- Added new ``cluster.routing.allocation.disk.watermark.flood_stage`` setting,
+  that controls at which disk usage indices should become read-only to prevent
+  running out of disk space. There is also a new node check that indicates
+  whether the threshold is exceeded.
+
+- Add ``max_token_length`` parameter to whitespace tokenizer.
+
+- Added new tokenizers ``simple_pattern`` and ``simple_pattern_split`` which
+  allow to tokenize text for the fulltext index by a regular expression
+  pattern.
+
+- Added support for CSV file inputs in ``COPY FROM`` statements. Input type is
+  inferred using the file's extension or can be set using the optional ``WITH``
+  clause and specifying the ``format``.
+
+- Fully qualified column names including a schema name will no longer match on
+  table aliases.
+
+- The default user if enterprise is disabled changed from ``null`` to
+  ``crate``. This causes entries in ``sys.jobs`` to show up with ``crate`` as
+  username. Functions like ``user`` will also return ``crate`` if enterprise is
+  enabled but the user module is not available.
+
+- Display the node information (name and id) of jobs in the ``sys.jobs`` table.
+
+- Changed the primary key constraints of the information schema tables
+  ``table_constraints``, ``referential_constraints``, ``table_partitions``,
+  ``key_column_usage``, ``columns``, and ``tables`` to be SQL compliant.
+
+- Arrays can now contain mixed types if they're safely convertible. JSON
+  libraries tend to encode values like ``[ 0.0, 1.2]`` as ``[ 0, 1.2 ]``, this
+  caused an error because of the strict type match we enforced before.
+
+- Implemented ``constraint_schema`` and ``table_schema`` in
+  ``information_schema.key_column_usage`` correctly and documented the full
+  table schema.
+
+- Statistics for jobs and operations are enabled by default. If you don't need
+  any statistics, please set ``stats.enabled`` to ``false``.
+
+- Changed ``BEGIN`` and ``SET SESSION`` to no longer require ``DQL``
+  permissions on the ``CLUSTER`` level.
+
+- Added 'epoch' argument to the extract function which returns the number of
```

Comment:
`epoch` could be seen as a keyword. inline code formatting would be appropriate.